performant-software/DM
# Add superuser to be granted all permissions on new projects; add example files for Digital Ocean deployment

### What does this PR do?
Adds a 'superuser' option to the configuration settings. If set, the superuser id will be available through a new 'store/users/superuser' endpoint, and this user will be granted all permissions on any new project created by another user.

Also adds placeholders in the machine-images/digitalocean.json configuration and in an application configuration example file, indicating where to set the Digital Ocean API token, OAuth provider settings, and superuser ID for a new deployment.

Also includes updates to the README regarding Digital Ocean deployment, and a minor fix to accommodate a newly required parameter for Google authentication.

### What issues does it address?
Closes #147 

### How to test
*Note: the superuser feature will need to be tested on the staging instance after all, since the real superuser has already been set on the new production instance*
Open the [test instance](http://45.55.63.32/) of DM that has been provisioned on Digital Ocean. Click Log In and either use a signed-in Google/Github account to authenticate, or create a new account at https://performant-oauth-provider.herokuapp.com/ . Once logged in, create a couple new projects. Then log out, go to https://performant-oauth-provider.herokuapp.com/, log out from there if currently signed in, and log in as the super user for this instance by entering username 'admin@example.com' and password 'password'. Return to [the DM instance](http://45.55.63.32/) and click Log In -> Performant. You should see the projects you created from the other account appear in the projects list. Open a project and check its access settings to confirm that you, as the super user, have full ability to change these and the published setting.

Additionally, go to the [new production instance](http://app.digitalmappa.org) and sign in. Confirm that projects and resources can be created and deleted and that functionality is maintained from the previous production instance.
